zerver/lib/events.py

```diff
@@ -551,6 +552,23 @@ def fetch_initial_state_data(
                 get_default_stream_groups(realm)
             )
 
+    if want("topic"):
+        # NOTE: As of March 2022 we're only returning pinned topics, which are visible to all users. If we
+        # start storing other topics, we might need to modify the query to account for what the user
+        # profile is allowed to see.
```

Comment:
Ah, I think we might already need to do this. What happens if there's a pinned topic in a private stream?

Also, are we okay sending up pinned topics for every stream the user has access to, even if they might not be subscribed?